### PR TITLE
Add basic pipelining support

### PR DESCRIPTION
**Pull request by dolzenko**
Very basic/hacked together quickly, not sure if this API is right, WDYT?

### CODE REVIEW

**Comment by dvirsky** [comment]
Sorry I didn't respond to your comment yesterday, but I had a (what I think is) more elegant API in mind:

```go
 func (c *RedisClient) AddMulti(rs ...AddRequest) ([]string, error) {
...
}
```

What do you think?

**Comment by dolzenko** [comment]
Sounds good, will update

**Comment by dvirsky** [comment]
Thanks ! Actually no need to make it variadic. just []AddRequest will do as the argument.

File: disque/disque.go [hunk 1]
```diff
@@ -111,14 +108,15 @@ func (c *RedisClient) Add(r AddRequest) (string, error) {
...
-	replies, err := redis.Values(c.conn.Do("")) //  flush the output buffer and receive pending replies
+// AddMulti sends multiple ADDJOB in pipeline
+func (c *RedisClient) AddMulti(rs []AddRequest) ([]string, error) {
+	for _, r := range rs {
+		if err := c.conn.Send("ADDJOB", addArgs(r)...); err != nil {
+			return nil, err
+		}
+	}
+	// flush the output buffer and receive pending replies
+	replies, err := redis.Values(c.conn.Do(""))
```
Comment (dvirsky) [inline disque/disque.go]: Isn't this usually done with flush/receive? never seen this pattern. but if it works - fine!

Comment (dvirsky) [inline disque/disque.go]: Also, why not use `redis.Strings` directly and not convert each response to string?

File: disque/disque_test.go [hunk 2]
```diff
@@ -63,12 +63,7 @@ func TestSendAdd(t *testing.T) {
 		},
 	}
 
-	err = client.SendAdd(ja)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	ids, err := client.FinishAdd()
+	ids, err := client.AddMulti([]AddRequest{ja})
```
Comment (dvirsky) [inline disque/disque_test.go]: maybe make sure the ids contain something valid?

File: disque/disque.go (hunk 1 above)
Comment (dolzenko) [inline disque/disque.go]: > Isn't this usually done with flush/receive? never seen this pattern. but if it works - fine!

Its a bit weird, yeah, but its in the docs https://godoc.org/github.com/garyburd/redigo/redis#hdr-Pipelining 
> If the command argument to the Do method is "", then the Do method will flush the output buffer and receive pending replies without sending a command.

Comment (dolzenko) [inline disque/disque.go]: > Also, why not use redis.Strings directly and not convert each response to string?

Sorry, not well versed in redigo approach, but ```
replies, err := redis.Strings(c.conn.Do(""))
```
gives me ```
disque_test.go:69: redigo: unexpected element type for Strings, got type string
```

**Comment by dvirsky** [comment]
Alright then, looks good. I'm merging!

**Comment by dvirsky** [comment]
thanks a lot

**Pull request merged**